sips/sip-temporary_keypairs-des-128.md

```diff
@@ -0,0 +1,123 @@
+         fs::write(path, encode_data)?;
+   and before new from files, keystore shoud try to decode from des-128:
+           if contents.starts_with("[") {
+                kp_strings = serde_json::from_str(&*contents)
+             .map_err(|e| anyhow!("Can't deserialize FileBasedKeystore from {:?}: {e}", path))?;
+            }else {
+                let decode_data = default_des_128_decode(contents);
+                kp_strings = serde_json::from_str(&*decode_data)
+                    .map_err(|e| anyhow!("Can't deserialize FileBasedKeystore from {:?}: {e}", path))?;
+            }
+2. for fullnode.yaml , network.yaml files, we should change the serder function
+   #[serde(default = "default_authority_key_pair")]
+   #[serde(serialize_with = "serialize_with_aes_encode_authoritykey")]
+   #[serde(deserialize_with = "deserialize_with_aes_encode_authoritykey")]
+   pub protocol_key_pair: AuthorityKeyPairWithPath,
+
+   #[serde(default = "default_key_pair")]
+   #[serde(serialize_with = "serialize_with_aes_encode")]
+   #[serde(deserialize_with = "deserialize_with_aes_encode")]
+   pub worker_key_pair: KeyPairWithPath,
+   #[serde(default = "default_key_pair")]
+   #[serde(serialize_with = "serialize_with_aes_encode")]
+   #[serde(deserialize_with = "deserialize_with_aes_encode")]
+   pub account_key_pair: KeyPairWithPath,
+   #[serde(default = "default_key_pair")]
+   #[serde(serialize_with = "serialize_with_aes_encode")]
+   #[serde(deserialize_with = "deserialize_with_aes_encode")]
+   pub network_key_pair: KeyPairWithPath,
+ 
+  the deserialize_with_aes_encode_xxx is a little complex, for we need to support both string and mapping type,
+  fn deserialize_with_aes_encode_authoritykey<'de, D>(deserializer: D) -> Result<AuthorityKeyPairWithPath, D::Error>
+  where
+  D: Deserializer<'de>,
+  {
+  let value: serde_yaml::Value = match serde_yaml::Value::deserialize(deserializer) {
+  Ok(value) => value,
+  Err(_) => return Err(D::Error::custom("Failed to deserialize value as YAML")),
+  };
+
+    match value {
+        serde_yaml::Value::String(mut s) => {
+            if s.starts_with(DEFAULT_AES_PREFIX) {
+                s = node_des_128_decode(s);
+            }
+            let keypair  = <AuthorityKeyPair as EncodeDecodeBase64>::decode_base64(&s);
+
+            Ok(AuthorityKeyPairWithPath::new(keypair.unwrap()))
+        }
+        serde_yaml::Value::Mapping(map) => {
+            let path = map.get(&Value::String("path".parse().unwrap())).ok_or_else(|| D::Error::custom("Missing path"))?;
+            println!("path: {:?}", path);
+            let keypair = read_authority_keypair_from_file(path.as_str().unwrap()).map_err(|e| D::Error::custom(format!("Failed to read keypair from file: {}", e)))?;
+            Ok(AuthorityKeyPairWithPath::new(keypair))
+        }
+        _ => {
+            Err(D::Error::custom("Invalid value type, expected string or mapping"))
+        }
+    }
+
+
+}
+
+
+## Rationale
```

Comment:
this section should provide the rationale behind the change such as 1) why encryption 2) why AES 3) why the prefix